06MPI/cpp/scatter.cc

```diff
@@ -5,17 +5,17 @@
 TEST_CASE("Collective communications") {
 
     int rank, size;
-    MPI_Comm_rank (MPI_COMM_WORLD, &rank);
-    MPI_Comm_size (MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
 
     SECTION("Collective scatter") {
       /// "scatter"
       std::string const message = "This message is going to come out in separate channels";
+      if (message.size() < (unsigned)size) return;
```

Comment:
In this case shouldn't you just make size unsigned so you don't need to do the cast here?

Reply:
Except the docs for MPI_Comm_size say it expects an `int`...

Reply:
I wonder if that is something you can file a bug with MPI for...

Reply:
Probably not :)